engine/nkiri.go

```diff
@@ -0,0 +1,209 @@
+	if len(titleSplit) >= 1 {
+		movie.Title = removeCaratRe.ReplaceAllString(titleSplit[0], "_")
+		movie.Category = strings.TrimSuffix(strings.TrimPrefix(titleSplit[1], "Download"), "Movie")
+	}
+	//Fetch UploadDate for ListMode Items
+	if engine.mode == ListMode {
+		movie.UploadDate = strings.TrimSpace(el.ChildText("div.blog-entry-date"))
+	}
+	//Fetch DownloadLink
+	downloadLink, err := url.Parse(el.ChildAttr("a", "href"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	movie.DownloadLink = downloadLink
+	if movie.Title != "" {
+		year := yearRe.FindStringSubmatch(movie.Title)
+		if len(year) > 1 {
+			intYear, err := strconv.Atoi(year[1])
+			if err == nil {
+				movie.Year = intYear
+			}
+		}
+	}
+	return movie, nil
+}
+
+func (engine *NkiriEngine) updateDownloadProps(downloadCollector *colly.Collector, movies *[]Movie) {
+	sizeRe := regexp.MustCompile(`(\d.*)`)
+	downloadCollector.OnHTML("div.elementor-section-wrap", func(e *colly.HTMLElement) {
+		movieIndex := getMovieIndexFromCtx(e.Request)
+		movie := &((*movies)[movieIndex])
+		seriesMap := map[string]*url.URL{}
+		episode := 0
+		nextSection := false
+		e.ForEach("section.elementor-section", func(n int, inner *colly.HTMLElement) {
+			switch {
+			//Fetch Download Link For Series
+			case strings.HasPrefix(inner.ChildText("span.elementor-button-text"), "Download Episode"):
+				episode++
+				downloadLink, err := url.Parse(inner.ChildAttr("div.elementor-button-wrapper > a", "href"))
+				if err != nil {
+					log.Fatal(err)
+				}
+				seriesMap[strconv.Itoa(episode)] = downloadLink
+			//Fetch DownloadLink For Movies
+			case strings.HasPrefix(inner.ChildText("span.elementor-button-text"), "Download Movie"):
+				downloadLink, err := url.Parse(inner.ChildAttr("div.elementor-button-wrapper > a", "href"))
+				if err != nil {
+					log.Fatal(err)
+				}
+				movie.DownloadLink = downloadLink
+			}
+			//Update movie size
+			if strings.HasPrefix(inner.ChildText("span.elementor-alert-title"), "Download Size") {
+				sizeMatch := sizeRe.FindStringSubmatch(
+					strings.TrimSpace(
+						inner.ChildText("span.elementor-alert-description")))
+				if len(sizeMatch) >= 1 {
+					movie.Size = strings.TrimSpace(sizeMatch[0])
+				}
+			}
+			//Fetch Movie Description
+			if strings.HasPrefix(inner.ChildText("div.elementor-container"), "Synopsis") {
+				nextSection = true
+			} else if nextSection {
+				movie.Description = inner.ChildText("p")
+				nextSection = false
+			}
+		})
+		if len(seriesMap) > 0 {
+			movie.IsSeries = true
+			movie.SDownloadLink = seriesMap
+		}
+	})
+}
+
+// List : list all the movies on a page
+func (engine *NkiriEngine) List(page int) SearchResult {
+	engine.mode = ListMode
+	result := SearchResult{
+		Query: "List of Recent Uploads - Page " + strconv.Itoa(page),
+	}
+	pageParam := fmt.Sprintf("page/%v", strconv.Itoa(page))
+	movies := []Movie{}
+	//Initialized List Categories
+	engine.ListCategories = []string{
+		"international",
+		"african",
```

Comment:
Why did you set it here.... Still set it up there when you're setting List and Search urls